sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/MetricFeedbackLiveTests.cs

```diff
@@ -0,0 +1,290 @@
+
+            if (populateOptionalMembers)
+            {
+                feedbackToAdd.StartTime = CreatedFeedbackStartTime;
+                feedbackToAdd.EndTime = CreatedFeedbackEndTime;
+            }
+
+            string feedbackId = await client.AddFeedbackAsync(feedbackToAdd);
+
+            Assert.That(feedbackId, Is.Not.Null);
+
+            MetricFeedback addedFeedback = await client.GetFeedbackAsync(feedbackId);
+
+            ValidateMetricFeedback(addedFeedback, feedbackId, dimensionKey);
+
+            Assert.That(addedFeedback.Type, Is.EqualTo(FeedbackType.Comment));
+
+            var commentFeedback = addedFeedback as MetricCommentFeedback;
+
+            Assert.That(commentFeedback, Is.Not.Null);
+            Assert.That(commentFeedback.Comment, Is.EqualTo(comment));
+
+            if (populateOptionalMembers)
+            {
+                Assert.That(commentFeedback.StartTime, Is.EqualTo(CreatedFeedbackStartTime));
+                Assert.That(commentFeedback.EndTime, Is.EqualTo(CreatedFeedbackEndTime));
+            }
+            else
+            {
+                Assert.That(commentFeedback.StartTime, Is.Null);
+                Assert.That(commentFeedback.EndTime, Is.Null);
+            }
+        }
+
+        [RecordedTest]
+        public async Task AddAndGetPeriodFeedback()
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var dimensionKey = new DimensionKey();
+            dimensionKey.AddDimensionColumn("city", "Delhi");
+            dimensionKey.AddDimensionColumn("category", "Handmade");
+
+            var filter = new FeedbackDimensionFilter(dimensionKey);
+            var periodValue = 10;
+
+            var feedbackToAdd = new MetricPeriodFeedback(MetricId, filter, PeriodType.AutoDetect, periodValue);
+
+            string feedbackId = await client.AddFeedbackAsync(feedbackToAdd);
+
+            Assert.That(feedbackId, Is.Not.Null);
+
+            MetricFeedback addedFeedback = await client.GetFeedbackAsync(feedbackId);
+
+            ValidateMetricFeedback(addedFeedback, feedbackId, dimensionKey);
+
+            Assert.That(addedFeedback.Type, Is.EqualTo(FeedbackType.Period));
+
+            var periodFeedback = addedFeedback as MetricPeriodFeedback;
+
+            Assert.That(periodFeedback, Is.Not.Null);
+            Assert.That(periodFeedback.PeriodType, Is.EqualTo(PeriodType.AutoDetect));
+            Assert.That(periodFeedback.PeriodValue, Is.EqualTo(periodValue));
+        }
+
+        [RecordedTest]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task GetAllFeedback(bool populateOptionalMembers)
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var options = new GetAllFeedbackOptions();
+
+            if (populateOptionalMembers)
+            {
+                options.TimeMode = FeedbackQueryTimeMode.FeedbackCreatedTime;
```

Comment:
do all optional values need to be set? or it really doesn't matter

Reply:
In general it doesn't matter, but there may be specific scenarios in which a "set" of options must be specified together. In this case, I would say it wouldn't make sense to specify a `StartTime` or an `EndTime` without a `TimeMode`, so that could possibly be a service-side error. I will create tests for these "invalid configurations" as part of this issue: https://github.com/Azure/azure-sdk-for-net/issues/17414.